bsip-0022.md

```diff
@@ -37,34 +37,202 @@ Currently within Bitshares DPOS system, an user's votes for committee/witness/pr
+|Effect of decay rate on referendum issues|Separate decay rates for every _decayable_ referendum category|Single decay rate shared by every referendum category|Single decay rate shared by every referendum category|
+|Effect of decay rate on other issues|None|Decays the share of blockchain profits that are received by an account|Decays the share of payment for standby block producers|
+
+## Software Specifications
+
+TODO
 
-* Rate of vote weight decay.
-  * Once the vote has aged past the 'max vote age' what length of time should pass before the vote is worth 0 BTS?
-* Max vote age (before votes begin to decay).
-  * Separate age variables for: Comittee, Witness & Proxy votes (proxies should potentially age at a slower rate than comittee/witness, as these users have opted to delegate their voting power to an active user).
-* Accounts blacklisted from voting (exchanges/services/scam accounts).
 
-## Decay function
+# Risks
 
-To keep things simple, we should start with a linear decay function. If this proves too aggressive/passive, the rate of decay could be modified or the formula could be changed at a later date to potentially be exponential.
+- [Vote decay](#vote-decay-function) can negatively impact voters (a) who lost the keys to their account but have been content with how their old votes were cast, (b) who are _unaware_ of its existence and who are expecting the effects of their votes to apply perpetually, or (c) who are _aware_ of its existence but cannot easily determine the age of their last vote.
 
-Example linear vote weight decay function: (note: pseudocode!)
-```
-//Assuming this function only triggers once the vote has passed the set 'max vote age' set by the committee, hence we don't check for the vote tx not having aged past the 'max_days_decay' in this example.
+- The effects of vote decay, especially during the decay of the [initially "grandfathered" votes](#initial-voting-power), may result in sudden changes to the elected witnesses, and committee members.  This sudden change could be  disruptive to those witnesses, and committee members that are suddenly voted out.
 
-let max_days_decay = 30; //Set by committee
-let start_decay_date = vote_date()
-let days_since_decay_began = todays_date() - ()
+- Core token holders who select [a proxy account that fails to re-cast its own votes periodically will have their effective voting power decay over time](#proxy-power) despite re-casting their own votes for that proxy.
 
-if (days_since_decay_began > max_days_decay) {
-    decaying_vote_weight = 0; //Or nullify the vote
-} else {
-    decaying_vote_weight = original_vote_weight - ((original_vote_weight/max_days_decay)*days_since_decay_began)
-}
-```
 
 # Discussion
 
```

Comment:
The calculations to be performed will significantly increase the duration of the maintenance interval, and with it replay time. I would like to see some numbers on the effect of the proposed changes assuming they had been activated at the beginning of our history (1. with the assumption that proxies are never updated, and 2. with the assumption that proxies are regularly updated).

Reply:
To avoid increasing of maintenance period, it would be better just assume that accounts with the last outgoing transaction(or other initiated by user operation) more than 1 year ago have zero vote power. imho.

If the last account activity was more than 1 year ago (for example), it means that the holder has died, lost his keys or simply doesn't participate in the community now, therefore we can't consider his voting power to be actual.

Reply:
> To avoid increasing of maintenance period, it would be better just assume that accounts with the last outgoing transaction(or other initiated by user operation) more than 1 year ago have zero vote power. imho.

I think that this suggestion can be an optimization that can avoid _some_ of the arithmetic overhead.

Reply:
> I would like to see some numbers on the effect of the proposed changes assuming they had been activated at the beginning of our history (1. with the assumption that proxies are never updated, and 2. with the assumption that proxies are regularly updated).

I will make sure that someone follows up on this.

Reply:
> To avoid increasing of maintenance period, it would be better just assume that accounts with the last outgoing transaction(or other initiated by user operation) more than 1 year ago have zero vote power.

I think this suggestion side-steps the original intent of this proposal. In my understanding, the idea is that users must actively participate in our governance model, by regularly updating their votes. Activity unrelated to voting should not prevent decay.